packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -846,9 +846,19 @@ abstract class SliderComponentShape {
 ///  * [SliderTrackShape] Base component for creating other custom track
 ///    shapes.
 class RectangularSliderTrackShape extends SliderTrackShape {
-  /// Abstract const constructor. This constructor enables subclasses to provide
-  /// const constructors so that they can be used in const expressions.
-  const RectangularSliderTrackShape();
+  /// Create a slider track that draws a circle 2 rectangles.
+  const RectangularSliderTrackShape({this.disabledThumbGap});
+
+  /// Spacing, or gap, between disabled thumb and the track.
+  ///
+  /// This is only used when the thumb is disabled. The material spec defaults
+  /// it to 2, which is half of its disabled thumb radius, which is 4.
+  final double disabledThumbGap;
+  double get _disabledThumbGap => disabledThumbGap ?? _defaultDisabledThumbGap;
```

Comment:
Since `_defaultDisabledThumbGap` is a const, I think it would be better to get rid of `_disabledThumbGap ` and instead set `disabledThumbGap = _defaultDisabledThumbGap` in the `RectangularSliderTrackShape` constructor.

Reply:
Done